src/components/date-picker/date-picker.js

```diff
@@ -0,0 +1,207 @@
+      allowInput: true,
+      dateFormat: 'm/d/Y',
+      mode: this.element.dataset.datePicker,
+      onClose: (selectedDates) => {
+        this._updateInputFields(selectedDates, type);
+      },
+      onChange: () => {
+        this._updateClassNames(calendar);
+      },
+      onMonthChange: () => {
+        this._updateClassNames(calendar);
+      },
+      onYearChange: () => {
+        this._updateClassNames(calendar);
+      },
+      onOpen: () => {
+        this._updateClassNames(calendar);
+      },
+      nextArrow: this._rightArrowHTML(),
+      prevArrow: this._leftArrowHTML(),
+    });
+    if (type === 'range') {
+      this.element.querySelector(this.options.selectorDatePickerInputTo).addEventListener('click', () => {
+        this.element.querySelector(this.options.selectorDatePickerInputTo).focus();
+        calendar.open();
+      });
+      this._addInputLogic(this.element.querySelector(this.options.selectorDatePickerInputTo));
+    }
+    this.element.querySelector(this.options.selectorDatePickerIcon).addEventListener('click', () => {
+      calendar.open();
+    });
+    this._addInputLogic(date);
+    return calendar;
+  }
+
+  getCalendar = () => this.element.calendar;
+
+  setMinDate = (date) => {
+    this.element.calendar.set('minDate', date);
+  }
+
+  setMaxDate = (date) => {
+    this.element.calendar.set('maxDate', date);
+  }
+
+  setDisabledDates = (dates) => {
+    this.element.calendar.set('disable', dates);
+  }
+
+  setDefaultDates = (dates) => {
+    this.element.calendar.set('defaultDate', dates);
+  }
+
+  _rightArrowHTML() {
+    return (`
+      <svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd">
+        <path d="M0 10.6L4.7 6 0 1.4 1.4 0l6.1 6-6.1 6z"></path>
+      </svg>`
+    );
+  }
+
+  _leftArrowHTML() {
+    return (`
+      <svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd">
+        <path d="M7.5 10.6L2.8 6l4.7-4.6L6.1 0 0 6l6.1 6z"></path>
+      </svg>`
+    );
+  }
+
+  _addInputLogic = (input) => {
+    const inputField = input;
+    let lastKey;
+    inputField.addEventListener('focus', () => {
+      inputField.value = inputField.value.replace(/\s+/g, '');
+    });
+    inputField.addEventListener('keydown', (e) => {
+      lastKey = e.which;
+    });
+    inputField.addEventListener('input', () => {
+      const noSlashes = (inputField.value.indexOf('/') === -1);
+      if (typeof this.element.dataset.datePickerMmYy === 'undefined') {
+        if ((inputField.value.length === 2 || inputField.value.length === 5) && lastKey !== 8) {
+          inputField.value += '/';
+        }
+      } else if (inputField.value.length === 2 && noSlashes && lastKey !== 8) {
+        inputField.value += '/';
+      }
+    });
+    inputField.addEventListener('blur', () => {
+      if ((inputField.value.length === 7 || inputField.value.length === 10) && inputField.value.indexOf('/') > -1) {
+        const inputValue = inputField.value;
```

Comment:
You shouldn't need to create a new variable for `inputValue`, you can just make the array variable:
```js
// const inputValue = inputField.value;
const inputValueArr = inputField.value.split('/');
```

And your `forEach` loop can use the `index` instead of the `increment` variable.
```js
inputValueArr.forEach((val, index) => {
  // increment++;
  if (index === inputValueArr.length) {
    newInputValue += `${val}`;
  } else {
    newInputValue += `${val} / `;
  }
});
```